holidays/ical.py

```diff
@@ -0,0 +1,234 @@
+        """
+        self.holidays = holidays_object
+        self.ical_timestamp = ical_timestamp or datetime.now(timezone.utc).strftime(
+            "%Y%m%dT%H%M%SZ"
+        )
+        self.holidays_version = __version__
+        self.language = self._validate_language(
+            getattr(self.holidays, "language", None)
+            or getattr(self.holidays, "default_language", None)
+            or "EN"
+        )
+
+    def _validate_language(self, language: str) -> str:
+        """
+        Validates the language code. Ensures it's a two-letter ISO 639-1 code
+        as iCal only supports that at the moment.
+
+        :param language:
+            The language code to validate.
+
+        :return:
+            Valid two-letter language code in uppercase.
+        """
+        language = language.upper().split("_")[0]
+
+        if len(language) != 2:
+            raise ValueError(
+                f"Invalid language code: {language}. "
+                "Only two-letter ISO 639-1 codes are supported by iCal. "
+                "Refer to the 'ISO 639-1 Code' column at "
+                "https://www.loc.gov/standards/iso639-2/php/code_list.php for valid codes."
+            )
+        return language
+
+    def _fold_line(self, line: str) -> str:
+        """
+        Fold long lines according to RFC 5545.
+
+        Content lines SHOULD NOT exceed 75 octets. If a line is too long,
+        it must be split into multiple lines, with each continuation line
+        starting with a space.
+
+        :param line:
+            The content line to be folded.
+
+        :return:
+            The folded content line.
+        """
+        if line.isascii():
+            # Simple split for ASCII: every (CONTENT_LINE_MAX_LENGTH - 1) chars,
+            # as first char of the next line is space
+            if len(line) > CONTENT_LINE_MAX_LENGTH:
+                return CONTENT_LINE_DELIMITER_WRAP.join(
+                    line[i : i + CONTENT_LINE_MAX_LENGTH - 1]
+                    for i in range(0, len(line), CONTENT_LINE_MAX_LENGTH - 1)
+                )
+
+        elif len(line.encode()) > CONTENT_LINE_MAX_LENGTH:
+            # Handle non-ASCII text while respecting byte length
+            parts = []
+            part_start = 0
+            part_len = 0
+            for i, char in enumerate(line):
+                char_byte_len = len(char.encode())
+                part_len += char_byte_len
+
+                if part_len > CONTENT_LINE_MAX_LENGTH:
+                    parts.append(line[part_start:i])
+                    part_start = i
+                    part_len = char_byte_len + 1  # line start with space
+
+            parts.append(line[part_start:])
+            return CONTENT_LINE_DELIMITER_WRAP.join(parts)
+
+        # Return as-is if it doesn't exceed the limit
+        return line
+
+    def _generate_event(self, date, holiday_name: str, holiday_length: int = 1) -> list[str]:
```

Comment:
```suggestion
    def _generate_event(self, dt: date, holiday_name: str, holiday_length: int = 1) -> list[str]:
```
(need `from datetime import date`)